tools/clang/include/clang/SPIRV/SpirvFunction.h

```diff
@@ -58,8 +74,15 @@ class SpirvFunction {
   void addBasicBlock(SpirvBasicBlock *);
 
 private:
-  QualType functionType;                    ///< This function's type
-  uint32_t functionId;                      ///< This function's <result-id>
+  uint32_t functionId; ///< This function's <result-id>
+
+  QualType returnType;        ///< The return type
```

Comment:
I'd suggest to rename them as `astReturnType` and `returnType`. The core should be on `SpirvType` instead of `QualType`.